pkg/provider/github/repository.go

```diff
@@ -0,0 +1,110 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-github/v47/github"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/v1alpha1"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/formatting"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/clients"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/templates"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ConfigureRepository(ctx context.Context, run *params.Run, req *http.Request, payload string, logger *zap.SugaredLogger) (bool, bool, error) {
+	// gitea set x-github-event too, so skip it for the gitea driver
```

Comment:
should we refactor that copy and paste code?

Reply:
what do you want to do here?

Reply:
isnt that the same code as the Detect function, should we have a common function that both uses?

Reply:
only

````
	if h := req.Header.Get("X-Gitea-Event-Type"); h != "" {
		return false, false, nil
	}
```
this is common, rest is changed